rust/agama-lib/src/storage/client/dasd.rs

```diff
@@ -0,0 +1,110 @@
+//! Implements a client to access Agama's D-Bus API related to DASD management.
+
+use zbus::{
+    fdo::ObjectManagerProxy,
+    zvariant::{ObjectPath, OwnedObjectPath},
+    Connection,
+};
+
+use crate::{
+    error::ServiceError,
+    storage::{model::dasd::DASDDevice, proxies::DASDManagerProxy},
+};
+
+/// Client to connect to Agama's D-Bus API for DASD management.
+#[derive(Clone)]
+pub struct DASDClient<'a> {
+    manager_proxy: DASDManagerProxy<'a>,
+    object_manager_proxy: ObjectManagerProxy<'a>,
+}
+
+impl<'a> DASDClient<'a> {
+    pub async fn new(connection: Connection) -> Result<DASDClient<'a>, ServiceError> {
+        let manager_proxy = DASDManagerProxy::new(&connection).await?;
+        let object_manager_proxy = ObjectManagerProxy::builder(&connection)
+            .destination("org.opensuse.Agama.Storage1")?
+            .path("/org/opensuse/Agama/Storage1")?
+            .build()
+            .await?;
+        Ok(Self {
+            manager_proxy,
+            object_manager_proxy,
+        })
+    }
+
+    pub async fn supported(&self) -> Result<bool, ServiceError> {
+        let introspect = self.manager_proxy.introspect().await?;
+        // simply check if introspection contain given interface
+        Ok(introspect.contains("org.opensuse.Agama.Storage1.DASD.Manager"))
+    }
+
+    pub async fn probe(&self) -> Result<(), ServiceError> {
+        Ok(self.manager_proxy.probe().await?)
+    }
+
+    pub async fn devices(&self) -> Result<Vec<(OwnedObjectPath, DASDDevice)>, ServiceError> {
+        let managed_objects = self.object_manager_proxy.get_managed_objects().await?;
+
+        let mut devices: Vec<(OwnedObjectPath, DASDDevice)> = vec![];
+        for (path, ifaces) in managed_objects {
+            if let Some(properties) = ifaces.get("org.opensuse.Agama.Storage1.DASD.Device") {
+                match DASDDevice::try_from(properties) {
+                    Ok(device) => {
+                        devices.push((path, device));
+                    }
+                    Err(error) => {
+                        log::warn!("Not a valid DASD device: {}", error);
+                    }
+                }
+            }
+        }
+        Ok(devices)
+    }
+
+    pub async fn format(&self, ids: &[&str]) -> Result<String, ServiceError> {
```

Comment:
API looks strange, it allows formating multiple DASDs but it return only one path for job.

Reply:
That's what I deduced from https://github.com/openSUSE/agama/blob/master/service/lib/agama/dbus/storage/interfaces/dasd_manager.rb#L139.

Reply:
there is only one job, that is ok, but then we need to check the status for each device, I think the missing part is the job Summary call on formatProgress callback.

https://github.com/openSUSE/agama/blob/master/web/src/client/storage.js#L891

Reply:
I say nothing, it already uses the Summary but the DASD path is not sent at all...

see ... https://github.com/openSUSE/agama/pull/1532/files#diff-481b9b86ced79110f138995e6097a0f0cde62016924c72f1c5c6397940566654R210
and ... https://github.com/openSUSE/agama/pull/1532/files#diff-481b9b86ced79110f138995e6097a0f0cde62016924c72f1c5c6397940566654R220